### PR TITLE
[Score API] Implement EngineScoreMixin for scoring functionality and refactor Tok…

### PR DESCRIPTION
**Pull request by sundar24295s**
### Motivation

The Scoring API (`/v1/score`, `Engine.score()`, `Engine.async_score()`) is a self-contained feature spanning entrypoints, managers, and tests. This PR extracts scoring-specific logic into dedicated files and adds CODEOWNERS entries so that scoring contributors are automatically requested for review on scoring-related changes.

#### 1. New file: `python/sglang/srt/entrypoints/engine_score_mixin.py`

Extracted `score()` and `async_score()` from the `Engine` class into an `EngineScoreMixin`. The `Engine` class now inherits from this mixin — no behavior change, just cleaner separation.

#### 2. Renamed: `tokenizer_manager_multiitem_mixin.py` → `tokenizer_manager_score_mixin.py`

Renamed the file and class (`TokenizerManagerMultiItemMixin` → `TokenizerManagerScoreMixin`) to better reflect its purpose. This file contains the core scoring logic: `score_request()`, `score_prompts()`, multi-item scoring helpers, and the `ScoreResult` dataclass.

#### 3. Updated: `python/sglang/srt/entrypoints/engine.py`

- `Engine` now inherits from `EngineScoreMixin` (MRO: `Engine → EngineScoreMixin → EngineBase → ABC`)
- Removed the inline `score()` / `async_score()` methods (now provided by the mixin)

#### 4. Updated: `.github/CODEOWNERS`

Added per-file ownership for all scoring-specific files:

```
/python/sglang/srt/entrypoints/engine_score_mixin.py @sundar24295s @chanh @fortunecookiee
/python/sglang/srt/entrypoints/openai/serving_score.py @sundar24295s @chanh @fortunecookiee
/python/sglang/srt/managers/tokenizer_manager_score_mixin.py @sundar24295s @chanh @fortunecookiee
/test/registered/core/test_score_api.py @sundar24295s @chanh @fortunecookiee
/benchmark/prefill_only/bench_score.py @sundar24295s @chanh @fortunecookiee
```

#### 5. Updated: `test/registered/openai_server/basic/test_serving_rerank.py`

Updated import path to reflect the renamed module.

### Files NOT Touched (and why)

These files contain scoring-related *infrastructure* (GPU kernels, scheduler logprob processing, attention masking) that is shared with other features. They were intentionally left in place:

- `layers/logits_processor.py` — `compute_logprobs_for_multi_item_scoring()` operates on GPU tensors using logits-processor-internal APIs
- `layers/attention/flashinfer_backend.py` — `MultiItemScoringParams` and `_process_multi_item_scoring()` are flashinfer attention-level code
- `managers/scheduler_output_processor_mixin.py` — multi-item scoring conditionals are interleaved with regular logprob processing
- `server_args.py` — `multi_item_scoring_delimiter` is a server config flag

#### Server startup

```bash
python -m sglang.launch_server \
  --model-path /shared/public/sharing/job-rank/kbehdin/f389cde308efd4dbb8d9-2025-06-06-18-31-30/best_model/epoch=0-step=498-HF \
  --port 30000 --host 0.0.0.0 \
  --chunked-prefill-size -1 \
  --enable-torch-compile \
  --dtype float16 \
  --max-prefill-tokens 100000 \
  --mem-fraction-static 0.3 \
  --enable-tokenizer-batch-encode \
  --disable-radix-cache \
  --disable-cuda-graph \
  --multi-item-scoring-delimiter 128255
```

Server starts successfully with no import errors.

#### Single-item scoring

```bash
curl -X POST "http://localhost:30000/v1/score" \
  -H "Content-Type: application/json" \
  -d '{
    "query": "What is the capital of California? Answer Yes or No for each of the following options:",
    "items": ["Scaramento"],
    "label_token_ids": [9454, 2753],
    "model": "..."
  }'
```

Response:
```json
{
    "scores": [[6.398421076647679e-06, 3.3389641633503636e-06]],
    "model": "...",
    "usage": {"prompt_tokens": 23, "total_tokens": 23},
    "object": "scoring"
}
```

#### Multi-item scoring (3 items, softmax)

```bash
curl -X POST "http://localhost:30000/v1/score" \
  -H "Content-Type: application/json" \
  -d '{
    "query": "What is the capital of California? Answer Yes or No for each of the following options:",
    "items": ["Sacramento", "San Jose", "San Francisco"],
    "label_token_ids": [9454, 2753],
    "apply_softmax": true,
    "model": "..."
  }'
```

Response:
```json
{
    "scores": [
        [0.4857216775417328, 0.5142783522605896],
        [0.6157812476158142, 0.3842187821865082],
        [0.5241511464118958, 0.475848913192749]
    ],
    "model": "...",
    "usage": {"prompt_tokens": 28, "total_tokens": 28},
    "object": "scoring"
}
```

#### Import verification

```python
from sglang.srt.entrypoints.engine import Engine
assert hasattr(Engine, 'score')
assert hasattr(Engine, 'async_score')
# MRO: Engine → EngineScoreMixin → EngineBase → ABC → object
```

### Test Plan

- [x] Server starts without import errors
- [x] `/v1/score` endpoint returns correct results for single-item scoring
- [x] `/v1/score` endpoint returns correct results for multi-item scoring with softmax
- [x] `Engine.score()` and `Engine.async_score()` are accessible via mixin inheritance
- [x] All Python imports resolve correctly (no circular imports)
- [x] No stale references to old file/class names remain in the codebase
- [ ] Existing CI test `test/registered/core/test_score_api.py` passes
- [ ] Existing CI test `test/registered/openai_server/basic/test_serving_rerank.py` passes

### CODE REVIEW

**Comment by hnyls2002** [comment]
All CI passed: https://github.com/sgl-project/sglang/actions/runs/23873600502/job/69759894670
